homeassistant/components/camera/push.py

```diff
@@ -0,0 +1,175 @@
+"""
+Camera platform that receives images through HTTP POST.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/camera.push/
+"""
+import logging
+
+from collections import deque
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.components.camera import Camera, PLATFORM_SCHEMA,\
+    STATE_IDLE, STATE_RECORDING
+from homeassistant.core import callback
+from homeassistant.components.http.view import HomeAssistantView
+from homeassistant.const import CONF_NAME, CONF_TIMEOUT, CONF_FORCE_UPDATE,\
+    HTTP_BAD_REQUEST
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.event import async_track_point_in_utc_time
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_BUFFER_SIZE = 'cache'
+CONF_IMAGE_FIELD = 'field'
+
+DEFAULT_NAME = "Push Camera"
+DEFAULT_FORCE_UPDATE = False
+
+ATTR_FILENAME = 'filename'
+ATTR_LAST_TRIP = 'last_trip'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_BUFFER_SIZE, default=1): cv.positive_int,
```

Comment:
maybe `queue_size` ? I'm not sure if  `buffer_size` to odd

Reply:
It's a buffer... that happens to be implemented with a queue. 

I don't think it makes sense to rename it.